core/task.rs

```diff
@@ -0,0 +1,131 @@
+// Copyright 2018-2023 the Deno authors. All rights reserved. MIT license.
+use core::pin::Pin;
+use core::task::Context;
+use core::task::Poll;
+use futures::Future;
+use std::marker::PhantomData;
+use tokio::runtime::Handle;
+use tokio::runtime::RuntimeFlavor;
+
+/// Equivalent to [`tokio::task::JoinHandle`].
+#[repr(transparent)]
+pub struct JoinHandle<R> {
+  handle: tokio::task::JoinHandle<MaskResultAsSend<R>>,
+  _r: PhantomData<R>,
+}
+
+impl<R> JoinHandle<R> {
+  /// Equivalent to [`tokio::task::JoinHandle::abort`].
+  pub fn abort(&self) {
```

Comment:
Do we ever use it? In LSP maybe?

Reply:
It was needed in a few places, but that was the only API required.